lib/test/t_IntegrationExpansion_std.cxx

```diff
@@ -0,0 +1,154 @@
+            // Check the coefficients
+            FunctionalChaosResult result(algo.getResult());
+            Point coeffs(result.getCoefficients().asPoint());
+            Point ref(coeffs.getSize());
+            std::copy(expectedCoefficientsLinear.begin(), expectedCoefficientsLinear.begin() + coeffs.getSize(), ref.begin());
+            Scalar err = (coeffs - ref).norm();
+            const Scalar rtol = 5.0e-2;
+            const Scalar atol = 5.0e-2;
+            assert_almost_equal(err, 0.0, rtol, atol);
+          }
+          if (wMin == wMax)
+            {
+              // Create the polynomial chaos algorithm using the full constructor
+              IntegrationExpansion algo(inputSample, outputSample, distribution, productBasis, basisSize);
+              algo.run();
+              // Check the coefficients
+              FunctionalChaosResult result(algo.getResult());
+              Point coeffs(result.getCoefficients().asPoint());
+              Point ref(coeffs.getSize());
+              std::copy(expectedCoefficientsLinear.begin(), expectedCoefficientsLinear.begin() + coeffs.getSize(), ref.begin());
+              Scalar err = (coeffs - ref).norm();
+              const Scalar rtol = 5.0e-2;
+              const Scalar atol = 5.0e-2;
+              assert_almost_equal(err, 0.0, rtol, atol);
+            }
+          {
+            // Create the polynomial chaos algorithm using the simplified constructor
+            IntegrationExpansion algo(inputSample, weights, outputSample, distribution);
+            algo.run();
+            // Check the coefficients
+            FunctionalChaosResult result(algo.getResult());
+            Point coeffs(result.getCoefficients().asPoint());
+            Point ref(coeffs.getSize());
+            std::copy(expectedCoefficientsHyper.begin(), expectedCoefficientsHyper.begin() + coeffs.getSize(), ref.begin());
+            Scalar err = (coeffs - ref).norm();
+            const Scalar rtol = 5.0e-2;
+            const Scalar atol = 5.0e-2;
+            assert_almost_equal(err, 0.0, rtol, atol);
+          }
+          // Check the constructors assuming uniform weights
+          if (wMin == wMax)
+            {
+              // Create the polynomial chaos algorithm using the simplified constructor
+              IntegrationExpansion algo(inputSample, outputSample, distribution);
+              algo.run();
+              // Check the coefficients
+              FunctionalChaosResult result(algo.getResult());
+              Point coeffs(result.getCoefficients().asPoint());
+              Point ref(coeffs.getSize());
+              std::copy(expectedCoefficientsHyper.begin(), expectedCoefficientsHyper.begin() + coeffs.getSize(), ref.begin());
+              Scalar err = (coeffs - ref).norm();
+              const Scalar rtol = 5.0e-2;
+              const Scalar atol = 5.0e-2;
+              assert_almost_equal(err, 0.0, rtol, atol);
+            } // Uniform weights
+        } // i
```

Comment:
There is no unit test of the `setActiveFunctions` here. It should be added, so that we can see that the method actually performs correctly.

Reply:
There is no call to the `setActiveFunctions` method in the unit test, so that we cannot make sure that the code will run. This check can be done _a minima_, without checkout the output. This will make sure that code runs. Improving the test can be done afterwards, but requires that the release at least evaluates the method in the unit tests.